File: 2419-longest-subarray-with-maximum-bitwise-and/2419-longest-subarray-with-maximum-bitwise-and.ts
function longestSubarray(nums: number[]): number {
  const max = Math.max(...nums);
  let ans = 0, contiguous = 0;
  for (const num of nums) {
    if (num === max) {
      if (++contiguous > ans) ans = contiguous;
    } else {
      contiguous = 0;
    }
  }
  return ans;
};

File: 2419-longest-subarray-with-maximum-bitwise-and/2419-longest-subarray-with-maximum-bitwise-and.go
import (
  "slices"
)

func longestSubarray(nums []int) int {
  maxNum := slices.Max(nums)
  ans, contiguous := 0, 0
  for _, num := range nums {
    if num == maxNum {
      contiguous++
      if contiguous > ans {
        ans = contiguous
      }
    } else {
      contiguous = 0
    }
  }
  return ans 
}